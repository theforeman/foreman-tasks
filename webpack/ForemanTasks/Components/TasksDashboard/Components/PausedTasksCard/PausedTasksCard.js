import React from 'react';
import { translate as __ } from 'foremanReact/common/I18n';

import TasksDonutCard from '../TasksDonutCard/TasksDonutCard';

const PausedTasksCard = ({ ...props }) => (
  <TasksDonutCard title={__('Paused')} {...props} />
);

const filterTitle = obj => {
  const { title, ...newObj } = obj;
  return newObj;
};

PausedTasksCard.propTypes = filterTitle(TasksDonutCard.propTypes);
PausedTasksCard.defaultProps = filterTitle(TasksDonutCard.defaultProps);

export default PausedTasksCard;
