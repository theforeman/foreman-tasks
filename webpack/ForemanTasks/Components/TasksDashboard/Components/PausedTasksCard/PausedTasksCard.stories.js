import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number, text, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withCardsDecorator } from '../../../../../stories/decorators';

import { TASKS_DONUT_CHART_FOCUSED_ON_OPTIONS_ARRAY } from '../TasksDonutChart/TasksDonutChartConstants';
import PausedTasksCard from './PausedTasksCard';

storiesOf('TasksDashboard', module)
  .addDecorator(withKnobs)
  .addDecorator(withCardsDecorator)
  .add('PausedTasksCard', () => (
    <PausedTasksCard
      last={number('last', 3)}
      older={number('older', 5)}
      timePeriod={text('timePeriod', '24h')}
      focusedOn={select(
        'focusedOn',
        TASKS_DONUT_CHART_FOCUSED_ON_OPTIONS_ARRAY,
        PausedTasksCard.defaultProps.focusedOn
      )}
      onTotalClick={action('onTotalClick')}
      onLastClick={action('onLastClick')}
      onOlderClick={action('onOlderClick')}
    />
  ));
