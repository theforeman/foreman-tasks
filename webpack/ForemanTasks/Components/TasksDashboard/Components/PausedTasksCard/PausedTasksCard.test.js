import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import PausedTasksCard from './PausedTasksCard';

const createRequiredProps = () => ({ last: 3, older: 5 });

const fixtures = {
  'render with minimal props': { ...createRequiredProps() },
};

describe('PausedTasksCard', () =>
  testComponentSnapshotsWithFixtures(PausedTasksCard, fixtures));
