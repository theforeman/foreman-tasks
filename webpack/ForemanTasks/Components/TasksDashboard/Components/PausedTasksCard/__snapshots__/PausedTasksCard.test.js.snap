// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PausedTasksCard render with minimal props 1`] = `
<TasksDonutCard
  className=""
  focusedOn="normal"
  last={3}
  older={5}
  title="Paused"
/>
`;
